src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -264,7 +264,16 @@ public Completable start() {
                 Transport transport = customTransport;
                 if (transport == null) {
                     Single<String> tokenProvider = negotiateResponse.getAccessToken() != null ? Single.just(negotiateResponse.getAccessToken()) : accessTokenProvider;
-                    switch (negotiateResponse.getChosenTransport()) {
+                    TransportEnum chosenTransport;
+                    if (this.skipNegotiate) {
+                        if (this.transportEnum != TransportEnum.WEBSOCKETS) {
+                            throw new RuntimeException("Negotiation can only be skipped when using the WebSocket transport directly with '.withTransport(TransportEnum.WEBSOCKETS)' on the 'HubConnectionBuilder'.");
+                        }
```

Comment:
You must have removed this from #30482 after I approved it. Could it break any previously-working code?

Reply:
Yeah, if you didn't specify a transport and used skip negotiate, it would work because WebSockets will be used if the server supports it. So adding that check would have broken people that had it working but didn't explicitly specify the transport.